### PR TITLE
Allow passing of preloaded requests to prepareData (fixes #50)

### LINKED ISSUE #50: injectPreparedData on the server side?
Can prepareData accept on optional `environment` so that injectPreparedData can be called on the server side if some data is known to already be available, such as from an upstream service?

If not provided, `environment` can still default to `new Relay.Environment()`.

If this seems like a reasonable approach, I can make a PR.

### PR DESCRIPTION
**Pull request by jakepusateri**
PR for #50 

Specifying environment allows injecting data already available locally. Specifying forceFetch allows `primeCache` to be used instead and a network request to not be needed if all data is present from preloading.

### CODE REVIEW

**Comment by jwb** [comment]
We need this capability.

File: src/prepareData.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 
-export default function prepareData({ Container, queryConfig }, networkLayer) {
+export default function prepareData({ Container, queryConfig, preloadedRequests = [] }, networkLayer) {
   return new Promise((resolve, reject) => {
     const environment = new Relay.Environment();
-
     const data = [];
 
+    const storeData = environment.getStoreData();
+    preloadedRequests.forEach(request => {
```
Comment (denvned) [inline src/prepareData.js]: Nice! But more ES6-ish syntax here would line better to the rest of the codebase:

``` js
for (const { query, response } of preloadedRequests) {
```

**Comment by denvned** [comment]
Thank you @jakepusateri!

**Comment by denvned** [comment]
Note that I changed `preloadedRequests` to be a third argument of `prepareData`.

This is now available in v0.7.3 of _isomorphic-relay_, and I also added `preloadedRequests` as a third argument to `prepareData` of [_isomorphic-relay-router_](https://github.com/denvned/isomorphic-relay-router) in v0.8.2.

**Pull request merged**